src/main/java/io/tiledb/spark/TileDBDataReaderPartitionScan.java

```diff
@@ -144,8 +175,8 @@ public boolean next() {
 
         // Compute the number of cells (records) that were returned by the query.
         HashMap<String, Pair<Long, Long>> queryResultBufferElements = query.resultBufferElements();
-        long currentNumRecords =
-            queryResultBufferElements.get(TILEDB_COORDS).getSecond() / domain.getNDim();
+        long currentNumRecords;
+        currentNumRecords = queryResultBufferElements.get(fieldNames.get(0)).getSecond();
```

Comment:
You can't do this because if field1 is a variable length attribute then the `second` parameter here will not represent the number of records. You need to do something like `if .getFirst() == valid; then .getFirst(); else .getSectnd();`

Reply:
Fixed.